content/docs/user-guide/programming/metrics/trace-event-logger.md

```diff
@@ -0,0 +1,293 @@
+        AZ::Metrics::CompleteArgs completeArgs;
+        completeArgs.m_name = "Complete Event";
+        completeArgs.m_cat = "Test";
+        completeArgs.m_dur = duration;
+        completeArgs.m_args = argsContainer;
+        completeArgs.m_id = "0";
+        auto resultOutcome = AZ::Metrics::RecordEvent(SampleLoggerId, AZ::Metrics::EventPhase::Complete, completeArgs);
+        EXPECT_TRUE(resultOutcome);
+    }
+
+    {
+        // Record Instant Event
+        AZ::Metrics::InstantArgs instantArgs;
+        instantArgs.m_name = "Instant Event";
+        instantArgs.m_cat = "Test";
+        instantArgs.m_args = argsContainer;
+        instantArgs.m_id = "0";
+        instantArgs.m_scope = AZ::Metrics::InstantEventScope::Thread;
+        auto resultOutcome = AZ::Metrics::RecordEvent(SampleLoggerId, AZ::Metrics::EventPhase::Instant, instantArgs);
+        EXPECT_TRUE(resultOutcome);
+    }
+
+    {
+        // Record Instant Event
+        // Add an extra object field for a count by making a copy of the argsContainer
+        auto extendedArgs = argsContainer;
+        extendedArgs.emplace_back("frameTime", AZ::Metrics::EventValue{ AZStd::in_place_type<AZ::s64>, 16 + threadIndex });
+        AZ::Metrics::CounterArgs counterArgs;
+        counterArgs.m_name = "Counter Event";
+        counterArgs.m_cat = "Test";
+        counterArgs.m_args = extendedArgs;
+        counterArgs.m_id = "0";
+        auto resultOutcome = AZ::Metrics::RecordEvent(SampleLoggerId, AZ::Metrics::EventPhase::Counter, counterArgs);
+        EXPECT_TRUE(resultOutcome);
+    }
+
+    {
+        // Record Async Start and End Events
+        // Also records the Async Instant event
+        constexpr AZStd::string_view asyncOuterEventName = "Async Event";
+
+        AZ::Metrics::AsyncArgs asyncArgs;
+        asyncArgs.m_name = asyncOuterEventName;
+        asyncArgs.m_cat = "Test";
+        asyncArgs.m_args = argsContainer;
+        asyncArgs.m_id = "0";
+        asyncArgs.m_scope = "Distinguishing Scope";
+        auto resultOutcome = AZ::Metrics::RecordEvent(SampleLoggerId, AZ::Metrics::EventPhase::AsyncStart, asyncArgs);
+        EXPECT_TRUE(resultOutcome);
+
+        // Change the name of the event
+        asyncArgs.m_name = "Async Instant Event";
+        resultOutcome = AZ::Metrics::RecordEvent(SampleLoggerId, AZ::Metrics::EventPhase::AsyncInstant, asyncArgs);
+        EXPECT_TRUE(resultOutcome);
+
+        // "Async Event" is the logical name of the being and end event being recorded
+        // So make sure the end event matches
+        asyncArgs.m_name = asyncOuterEventName;
+        resultOutcome = AZ::Metrics::RecordEvent(SampleLoggerId, AZ::Metrics::EventPhase::AsyncEnd, asyncArgs);
+        EXPECT_TRUE(resultOutcome);
+    }
+
+    // Flush and closes the event stream
+    // This completes the json array
+    googleTraceLogger.ResetStream(nullptr);
+}
```

Comment:
I think it would be better to come up with the shortest example, the system needs instead of unit tests examples, which are fairly large for an educational example.

Reply:
The recording metrics example has been simplified